impl/src/cargo_config.rs

```diff
@@ -0,0 +1,18 @@
+use serde::Deserialize;
+use std::collections::BTreeMap;
+
+/// A representation of the parts of .cargo/config.toml files understood by cargo-raze.
+/// Unknown fields are set to deny, as cargo-raze does not know how to interpret them,
```

Comment:
Am I understanding correctly that this rejects all fields except for `registries`?

We don't control the specification of `.cargo/config.toml` and shouldn't be panicing or returning errors if new fields are added here right? What's the rationale for erroring here?

Reply:
Yeah, this was me being overly defensive in case there are things in the .cargo/config.toml that would be problematic for raze (say, a config setting which referenced a relative path, or which changed some semantics in some way), but it's probably not actually appropriate for raze - easier for folks to file bugs if raze is broken by a setting they use, than to need to manually support every non-breaking feature :)

I've removed the check